packages/driver/src/cy/commands/sessions.ts

```diff
@@ -0,0 +1,822 @@
+import stringifyStable from 'json-stable-stringify'
+import * as $stackUtils from '../../cypress/stack_utils'
+const currentTestRegisteredSessions = new Map()
+const getSessionDetails = (sessState) => {
+  return {
+    id: sessState.id,
+    data: _.merge(
+      _.mapValues(_.groupBy(sessState.cookies, 'domain'), (v) => ({ cookies: v.length })),
+      ..._.map(sessState.localStorage, (v) => ({ [$Location.create(v.origin).hostname]: { localStorage: Object.keys(v.value).length } })),
+    ) }
+}
+const getSessionDetailsForTable = (sessState) => {
+  return _.merge(
+    _.mapValues(_.groupBy(sessState.cookies, 'domain'), (v) => ({ cookies: v })),
+    ..._.map(sessState.localStorage, (v) => ({ [$Location.create(v.origin).hostname]: { localStorage: v } })),
+  )
+}
+
+const getConsoleProps = (sessState) => {
+  const ret = {
+    id: sessState.id,
+    table: _.compact(_.flatMap(getSessionDetailsForTable(sessState), (val, domain) => {
+      return [() => {
+        return {
+          name: `🍪 Cookies - ${domain} (${val.cookies.length})`,
+          data: val.cookies,
+        }
+      },
+      val.localStorage && (() => {
+        return {
+          name: `📁 Storage - ${domain} (${_.keys(val.localStorage.value).length})`,
+          data: _.map(val.localStorage.value, (value, key) => {
+            return {
+              key, value,
+            }
+          }),
+        }
+      })]
+    }))
+    ,
+  }
+
+  return ret
+}
+
+export default function (Commands, Cypress, cy) {
+  const { Promise } = Cypress
+
+  const setActiveSession = (obj) => {
+    const currentSessions = cy.state('activeSessions') || {}
+
+    const newSessions = { ...currentSessions, ...obj }
+
+    cy.state('activeSessions', newSessions)
+  }
+  const getActiveSession = (id) => {
+    const currentSessions = cy.state('activeSessions') || {}
+
+    return currentSessions[id]
+  }
+  const clearActiveSessions = () => {
+    const curSessions = cy.state('activeSessions') || {}
+
+    cy.state('activeSessions', _.mapValues(curSessions, (v) => ({ ...v, hydrated: false })))
+  }
+
+  async function mapOrigins (origins) {
+    const current_origin = $Location.create(window.location.href).origin
+
+    return _.uniq(
+      _.flatten(await Promise.map(
+        ([] as string[]).concat(origins), async (v) => {
+          if (v === '*') {
+            return _.keys(await Cypress.backend('get:renderedHTMLOrigins')).concat([current_origin])
+          }
+
+          if (v === 'current_origin') return current_origin
+
+          return $Location.create(v).origin
+        },
+      )),
+    ) as string[]
+  }
+
+  async function _setStorageOnOrigins (originOptions) {
+    const specWindow = cy.state('specWindow')
+
+    const current_origin = $Location.create(window.location.href).origin
+
+    const current_origin_options_index = _.findIndex(originOptions, { origin: current_origin })
+
+    if (current_origin_options_index !== -1) {
```

Comment:
This is a minor nitpick, but why are these snake-cased? We use camel-casing for variables everywhere else in the codebase.

Reply:
85d85eaeeb